pkg/aws/cloudformation.go

```diff
@@ -114,83 +115,67 @@ func (c *awsClient) EnsureOsdCcsAdminUser(stackName string, adminUserName string
 
 func (c *awsClient) CreateStack(cfTemplateBody, stackName string) (bool, error) {
 	// Create cloudformation stack
-	_, err := c.cfClient.CreateStack(buildCreateStackInput(cfTemplateBody, stackName))
+	_, err := c.cfClient.CreateStack(context.Background(), buildCreateStackInput(cfTemplateBody, stackName))
 	if err != nil {
 		return false, err
 	}
 
-	// Wait until cloudformation stack creates
-	err = c.cfClient.WaitUntilStackCreateComplete(&cloudformation.DescribeStacksInput{
-		StackName: aws.String(stackName),
-	})
+	err = waitForStackCreateComplete(context.Background(), c.cfClient, stackName)
 	if err != nil {
-		switch typed := err.(type) {
-		case awserr.Error:
-			// Waiter reached maximum attempts waiting for the resource to be ready
-			if typed.Code() == request.WaiterResourceNotReadyErrorCode {
```

Comment:
Should we keep the error type check and the logging for the max retries scenario?

Reply:
Since we are only logging the error in case of this error type, and there isn't a special handling for that, I think it's okay to omit it with V2 if there isn't an equivalent error code.